### PR TITLE
[SPARK-42303][SQL] Assign name to _LEGACY_ERROR_TEMP_1326

### PR DESCRIPTION
**Pull request by itholic**
### What changes were proposed in this pull request?

This PR proposes to assign name to _LEGACY_ERROR_TEMP_1326, "CANNOT_MODIFY_CONFIG".

### Why are the changes needed?

We should assign proper name to _LEGACY_ERROR_TEMP_*

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

`./build/sbt "sql/testOnly org.apache.spark.sql.SQLQueryTestSuite*"`

### CODE REVIEW

**Comment by itholic** [comment]
Another top LEGACY error message. cc @srielau @MaxGekk @cloud-fan

File: core/src/main/resources/error/error-classes.json
Comment (MaxGekk) [inline core/src/main/resources/error/error-classes.json:82]: This one clashes with https://github.com/apache/spark/pull/39820, correct?

File: core/src/main/resources/error/error-classes.json
Comment (MaxGekk) [inline core/src/main/resources/error/error-classes.json:81]: nit:
```suggestion
      "Cannot modify the value of the Spark config: <key>.",
```

**Review by MaxGekk — APPROVED** [review]
Waiting for CI.

File: core/src/main/resources/error/error-classes.json
Comment (itholic) [reply core/src/main/resources/error/error-classes.json:82]: Oh... yes.
Will keep this in mind fix this in #39820 after merging.

Comment (MaxGekk) [reply core/src/main/resources/error/error-classes.json:82]: It has been merged, @itholic please, adjust this PR.

Comment (itholic) [reply core/src/main/resources/error/error-classes.json:82]: Got it! Thanks for the notify

Comment (itholic) [reply core/src/main/resources/error/error-classes.json]: Adjusted, thanks!

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master/3.4.
Thank you, @itholic.

**Pull request closed without merging**
